### PR TITLE
Adding a function to get display-name for a region.

### PR DESCRIPTION
**Pull request by snktd**

### CODE REVIEW

File: experimental/displaynames/src/options.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    /// The language display kind, defaults to "dialect".
+    pub language_display: LanguageDisplay,
+}
+
+/// An enum for formatting style.
+#[allow(missing_docs)] // The variants are self explanotory.
+#[non_exhaustive]
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum Style {
+    Narrow,
```
Comment (sffc) [inline experimental/displaynames/src/options.rs:47]: Note: No Narrow data for regions, but there is Short data, and Narrow data maybe for other types

File: experimental/displaynames/src/displaynames.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+impl DisplayNames {
+    /// Creates a new [`DisplayNames`] from locale data and an options bag.
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    /// <div class="stab unstable">
+    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
+    /// </div>
+    pub fn try_new_unstable<D: DataProvider<TerritoryDisplayNamesV1Marker> + ?Sized>(
```
Comment (sffc) [inline experimental/displaynames/src/displaynames.rs:47]: `try_new_region_unstable`

Comment (sffc) [inline experimental/displaynames/src/displaynames.rs:47]: Note: Consider changing all references to "territory" to "region", more consistent with 402. @markusicu ?

File: experimental/displaynames/src/options.rs (hunk 1 above)
Comment (snktd) [reply experimental/displaynames/src/options.rs:47]: There is no Long data in CLDR for territory displaynames. I am keeping these options as is for other types.

File: experimental/displaynames/src/displaynames.rs (hunk 2 above)
Comment (snktd) [reply experimental/displaynames/src/displaynames.rs]: Done.

Comment (snktd) [reply experimental/displaynames/src/displaynames.rs]: I plan to add a new PR for this change.

File: experimental/displaynames/src/displaynames.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                locale,
+                metadata: Default::default(),
+            })?
+            .take_payload()?;
+        Ok(Self { options, data })
+    }
+
+    /// Returns the display name of the region for a given string.
+    /// This function is locale-sensitive.
+    pub fn of(&self, region_code: &str) -> Result<&str, TinyStrError> {
```
Comment (sffc) [inline experimental/displaynames/src/displaynames.rs:63]: Note: When we add Language Names, the signature for this function will be more clear

**Review by sffc — APPROVED** [review]

**Review by sffc — DISMISSED** [review]
LGTM; please make CI happy.

**Comment by snktd** [comment]
> 

This is done and ready to merge now.

**Review by sffc — APPROVED** [review]

**Pull request merged**